pkg/kclient/ingress.go

```diff
@@ -3,15 +3,41 @@ package kclient
 import (
 	"context"
 	"fmt"
-
+	"github.com/openshift/odo/pkg/unions"
 	"github.com/pkg/errors"
 
 	extensionsv1 "k8s.io/api/extensions/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// CreateIngress creates an ingress object for the given service and with the given labels
-func (c *Client) CreateIngress(ingress extensionsv1.Ingress) (*extensionsv1.Ingress, error) {
+func (c *Client) CreateIngress(ingress unions.KubernetesIngress) (*unions.KubernetesIngress, error) {
+	var err error
+	if !ingress.IsGenerated() {
+		return nil, fmt.Errorf("please pass a generated ingress")
+	}
+	created := false
+	kii := unions.NewEmptyKubernetesIngress()
+	if c.isNetworkingV1IngressSupported {
+		kii.NetworkingV1Ingress, err = c.KubeClient.NetworkingV1().Ingresses(c.Namespace).Create(context.TODO(), ingress.NetworkingV1Ingress, metav1.CreateOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("failed to create networking v1 ingress %w", err)
```

Comment:
Can you use errors.Wrap instead? (in other places too)

Reply:
AFAIK `errors.Wrapf` was effectively rendered obsolete by the introduction of `%w` in `fmt.Errorf`